src/pages/api/auth/twitter/index.page.ts

```diff
@@ -0,0 +1,52 @@
+import * as admin from 'firebase-admin'
+import { paginateDepleteProjectsQueryCall } from 'lib/apollo/paginateDepleteProjectsQuery'
+import { consolidateMetadata } from 'models/project-metadata'
+import { NextApiRequest, NextApiResponse } from 'next'
+import { firestoreAdmin } from 'utils/firebase/firebaseAdmin'
+import { findProjectMetadata } from 'utils/server/ipfs'
+
+interface ApiRequest extends NextApiRequest {
+  body: {
+    projectId: number
+    cv: string
+  }
+}
+
+const handler = async (req: ApiRequest, res: NextApiResponse) => {
+  try {
+    const { projectId, cv } = req.body
+    const projects = await paginateDepleteProjectsQueryCall({
+      variables: {
+        where: { cv, projectId },
+        first: 1,
+      },
+    })
+    if (!projects.length) {
+      return res.status(404).json({ message: 'Project not found' })
+    }
+    const metadata = await findProjectMetadata({
+      metadataCid: projects[0].metadataUri!,
+    })
+    const { twitter } = consolidateMetadata(metadata)
+    if (!twitter) {
```

Comment:
**question** we already know the Twitter handle on the client-side. Could we send the Twitter handle as part of the request body (along side projectId and cv) to save us doing extra network requests here?

Reply:
If the Twitter handle can be sent in the request body, then any user could verify any Twitter handle for the project

ie:

- I own twitter handle `MischeviousBoi`
- I make project 420, a fake Juicebox, set handle to `@Juicebox`
- I post {projectId: 420, username: 'MischeviousBoi'}
- Project 1 shows the verified checkmark for handle `@Juicebox` in the social links

Reply:
Only posting project ID and having the backend do a contract read to get the set handle from the metadata feels the most secure to me.